File: infra/src/main/java/com/beat/infra/persistence/staff/mapper/StaffPersistenceMapper.java
package com.beat.infra.persistence.staff.mapper;

import org.springframework.stereotype.Component;

import com.beat.domain.performance.model.Staff;
import com.beat.infra.persistence.staff.entity.StaffJpaEntity;

@Component
public class StaffPersistenceMapper {

	public Staff toDomain(StaffJpaEntity entity) {
		return Staff.rehydrate(
			entity.getId(),
			entity.getStaffName(),
			entity.getStaffRole(),
			entity.getStaffPhoto()
		);
	}

	public StaffJpaEntity toEntity(Staff staff, long performanceId) {
		return StaffJpaEntity.rehydrate(
			staff.getId(),
			staff.getStaffName(),
			staff.getStaffRole(),
			staff.getStaffPhoto(),
			performanceId
		);
	}
}

File: infra/src/main/kotlin/com/beat/infra/persistence/staff/mapper/StaffPersistenceMapper.kt
package com.beat.infra.persistence.staff.mapper

import com.beat.domain.performance.model.Staff
import com.beat.infra.persistence.staff.entity.StaffJpaEntity
import org.springframework.stereotype.Component

@Component
class StaffPersistenceMapper {
    fun toDomain(entity: StaffJpaEntity): Staff = Staff.rehydrate(
        entity.id,
        entity.staffName,
        entity.staffRole,
        entity.staffPhoto,
    )

    fun toEntity(staff: Staff, performanceId: Long): StaffJpaEntity = StaffJpaEntity.rehydrate(
        staff.getId(),
        staff.staffName,
        staff.staffRole,
        staff.staffPhoto,
        performanceId,
    )
}
